TestUtil.java

```diff
@@ -0,0 +1,270 @@
+	 * @param lectures The number of "lectures" from the start to end date
+	 * @param start    The date of the first lecture.
+	 * @param stop     A date after the last lecture.
+	 * @param netIDs   A list of all netIDs in the roster.
+	 * @param dict     A list of words to be used in generating feedback.
+	 * @param config   A double array of size 6 whose elements are all in the
+	 *                 range [0,1] that specifies the probabilities of various
+	 *                 events.
+	 * @return
+	 */
+	public static ArrayList<String> 
+	    generateFeedback(int lectures, Date start, Date stop, 
+			             ArrayList<String> netIDs, ArrayList<String> dict,
+			             double[] config)
+	{
+		SimpleDateFormat form = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
+		long range = stop.getTime() - start.getTime();
+		long interval = range/lectures;
+		//config = {ATTENDANCE, VALID_PARTNER, HAS_FEEDBACK, 
+		//          STRENGTH_P_VAL, WEAKNESS_P_VAL, DUPLICATE};
+		long currentTime = start.getTime();
+		int numStudents = netIDs.size();
+		ArrayList<String> products = new ArrayList<>();
+		while (lectures-- > 0){
+			//Forgive this monstrosity. Excessive indentation makes me cry.
+			for (String person : netIDs) if (Math.random() < config[0]) do{
+				long time = currentTime + (long) (Math.random()*interval);
+				String date = form.format(new Date(time));
+				String partner = Math.random() < config[1] ?
+						netIDs.get(gen.nextInt(numStudents)) :
+						"null";
+				int rating = 1 + gen.nextInt(10);
+				boolean writeResponse = Math.random() < config[2];
+				String strength = writeResponse ?
+					randomSentence(dict, randomGeometric(config[3])) :
+				    "";
+				String weakness = writeResponse ?
+					randomSentence(dict, randomGeometric(config[4])) :
+					"";
+				String product = String.format(
+				    "\"%s\", \"%s\", \"%d\", \"%s\", \"%s\", \"%s\"",
+				    person, partner, rating, strength, weakness, date);
+				products.add(product);
+			}while(Math.random() < config[5]);
+			currentTime += interval;
+		}
+		return products;
+	}
+	
+	
+	/**
+	 * Takes in a collection of PeerInteraction objects and generates an
+	 * ArrayList containing all the words used in strength and weakness fields
+	 * of all elements of the collection. This is simply used to later generate
+	 * fake PeerInteraction objects.
+	 * 
+	 * @param samples A collection of PeerInteractions 
+	 * @return An ArrayList containing all distinct elements from the argument.
+	 */
+	public static ArrayList<String> 
+	    dictFromInteractions(Iterable<PeerInteraction> samples)
+	{
+		ArrayList<String> output = new ArrayList<>();
+		for (PeerInteraction entry : samples)
+			if (entry.hasFeedback()){
+				String cat = entry.getStrength() + ' ' + entry.getWeakness();
+				Scanner wordExtractor = new Scanner(cat);
+				while (wordExtractor.hasNext())
+					output.add(wordExtractor.next().replace("\"", "\"\""));
+				wordExtractor.close();
+			}
+		output.sort(naturalOrder);
```

Comment:
I think this is because Travis uses Java SE 7? I'll change my local compiler and fix this. I would recommend compiling the project Java SE 8 though. It's a large improvement over SE 7 in terms of features (eg lambdas).